src/background/extension.ts

```diff
@@ -33,6 +31,9 @@ export class Extension {
     tabs: TabManager;
     user: UserStorage;
 
+    private isEnabledNow: boolean = null;
+
+    static ALARM_NAME = 'auto-time-alarm';
```

Comment:
`ALARM_NAME` seems a generic name.

As an idea, maybe try using 2 alarms? One for the beginning of time interval and another one for the ending?

Reply:
> `ALARM_NAME` seems a generic name.

I had to create internal alarms for Newsmaker, Network, and Extension classes, so I called them `Newsmaker.ALARM_NAME`, `LimitedCacheStorage.ALARM_NAME`, and `Extension.ALARM_NAME` as a naming convention. If you have a better suggestion, I can rename them and one in #6327. (I just need static string variables with unique names, but their names and even values can be entirely arbitrary.)

> As an idea, maybe try using 2 alarms? One for the beginning of time interval and another one for the ending?

I'm not sure how that would be an improvement, could you elaborate? These alarms can't be recurring alarms anyway because the exact sunset/dawn times vary a bit.